integrations/hanlp/src/haystack_integrations/components/preprocessors/hanlp/chinese_document_splitter.py

```diff
@@ -0,0 +1,499 @@
+    Therefore, splitting by word means splitting by these multi-character tokens,
+    not simply by single characters or spaces.
+
+    ### Usage example
+    ```python
+    doc = Document(content=
+        "这是第一句话，这是第二句话，这是第三句话。"
+        "这是第四句话，这是第五句话，这是第六句话！"
+        "这是第七句话，这是第八句话，这是第九句话？"
+    )
+
+    splitter = ChineseDocumentSplitter(
+        split_by="word", split_length=10, split_overlap=3, respect_sentence_boundary=True
+    )
+    splitter.warm_up()
+    result = splitter.run(documents=[doc])
+    print(result["documents"])
+    ```
+    """
+
+    def __init__(
+        self,
+        split_by: Literal["word", "sentence", "passage", "page", "line", "period", "function"] = "word",
+        split_length: int = 1000,
+        split_overlap: int = 200,
+        split_threshold: int = 0,
+        respect_sentence_boundary: bool = False,
+        splitting_function: Optional[Callable] = None,
+        granularity: Literal["coarse", "fine"] = "coarse",
+    ):
+        """
+        Initialize the ChineseDocumentSplitter component.
+
+        :param split_by: The unit for splitting your documents. Choose from:
+            - `word` for splitting by spaces (" ")
+            - `period` for splitting by periods (".")
+            - `page` for splitting by form feed ("\\f")
+            - `passage` for splitting by double line breaks ("\\n\\n")
+            - `line` for splitting each line ("\\n")
+            - `sentence` for splitting by HanLP sentence tokenizer
+
+        :param split_length: The maximum number of units in each split.
+        :param split_overlap: The number of overlapping units for each split.
+        :param split_threshold: The minimum number of units per split. If a split has fewer units
+            than the threshold, it's attached to the previous split.
+        :param respect_sentence_boundary: Choose whether to respect sentence boundaries when splitting by "word".
+            If True, uses HanLP to detect sentence boundaries, ensuring splits occur only between sentences.
+        :param splitting_function: Necessary when `split_by` is set to "function".
+            This is a function which must accept a single `str` as input and return a `list` of `str` as output,
+            representing the chunks after splitting.
+        :param granularity: The granularity of Chinese word segmentation, either 'coarse' or 'fine'.
+
+        :raises ValueError: If the granularity is not 'coarse' or 'fine'.
+        """
+        self.split_by = split_by
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_threshold = split_threshold
+        self.respect_sentence_boundary = respect_sentence_boundary
+        self.splitting_function = splitting_function
+        self.granularity = granularity
+
+        if granularity not in {"coarse", "fine"}:
+            msg = f"Invalid granularity '{granularity}'. Choose either 'coarse' or 'fine'."
+            raise ValueError(msg)
+
```

Comment:
We can add a method to check if the parameters used for splitting are valid, e.g.:

```python
def __post_init__(self):
    if self.split_length <= 0:
        raise ValueError("split_length must be positive")
    if self.split_overlap < 0:
        raise ValueError("split_overlap must be non-negative")
    if self.split_overlap >= self.split_length:
        raise ValueError("split_overlap must be less than split_length")

```